File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.py
# 모든 노드의 최단 거리 -> 플로이드 워셜
n, m = map(int, input().split())

arr = [list(map(int, input().split())) for _ in range(n)]

for i in range(n):
    for a in range(n):
        for b in range(n):
            arr[a][b] = min(arr[a][b],arr[a][i] + arr[i][b])

for _ in range(m):
    a, b, c = map(int, input().split())
    # 1에서 1로 가는게 0,0인덱스이므로 a-1,b-1
    # c시간내로 올 수 있다면
    if arr[a-1][b-1] <= c:
        print('Enjoy other party')
    else:
        print('Stay here')

File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.js
const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : fs.readFileSync("example.txt").toString().split("\r\n");

const [n,m] = input[0].split(" ").map((v)=>parseInt(v));

let arr = new Array();

for (let i=0; i< n; i++) {
  const ar = input[i+1].split(" ").map((v)=> parseInt(v));
  arr.push(ar)
}


for (let k=0; k < n; k++) {
  for (let i =0; i <n; i++) {
    for (let j = 0; j < n; j++) {
      arr[i][j] = Math.min(arr[i][j], arr[i][k] + arr[k][j])
    }
  }
}

for (let i = 0; i <m; i++) {
  const [a,b,c] = input[i+1+n].split(" ").map((v)=>parseInt(v))
  if (arr[a-1][b-1] <= c) {
    console.log('Enjoy other party')
  } else {
    console.log('Stay here')
  }
}